app/controllers/assumed/create/CheckYourAnswersController.scala

```diff
@@ -79,36 +77,32 @@ class CheckYourAnswersController @Inject()(override val messagesApi: MessagesApi
       Ok(view(list, operatorId))
   }
 
-  def onSubmit(operatorId: String): Action[AnyContent] = (identify andThen checkAssumedReportingAllowed andThen getData(operatorId) andThen requireData).async {
-    implicit request =>
+  def onSubmit(operatorId: String): Action[AnyContent] =
+    (identify andThen checkAssumedReportingAllowed andThen getData(operatorId) andThen requireData).async { implicit request =>
       getAnswerAsync(PlatformOperatorSummaryQuery) { platformOperator =>
-
-        userAnswersService.toAssumedReportingSubmission(request.userAnswers)
-          .map(Future.successful)
-          .left.map(errors => Future.failed(BuildAssumedReportingSubmissionFailure(errors)))
-          .merge
-          .flatMap { submissionRequest =>
-            (for {
-              submission <- submit(submissionRequest, platformOperator)
-              summary <- AssumedReportSummary(request.userAnswers).map(Future.successful).getOrElse(Future.failed(Exception("unable to build an assumed report summary")))
-              emptyAnswers = UserAnswers(request.userId, operatorId, Some(summary.reportingPeriod))
-              emailsSentResult <- emailService.sendAddAssumedReportingEmails(platformOperator, submission.created, summary)
-              answersWithSummary <- Future.fromTry(emptyAnswers.set(AssumedReportSummaryQuery, summary))
-              answersWithEmailSentResult <- Future.fromTry(answersWithSummary.set(SentAddAssumedReportingEmailsQuery, emailsSentResult))
-              _ <- sessionRepository.set(answersWithEmailSentResult)
-              _ <- sessionRepository.clear(request.userId, operatorId, None)
-            } yield Redirect(routes.SubmissionConfirmationController.onPageLoad(operatorId, summary.reportingPeriod))).recover {
-              case error: SubmitAssumedReportingFailure => logger.warn("Failed to submit assumed reporting", error)
-                throw error
-            }
+        CreateAssumedReportingSubmissionRequest.build(request.userAnswers).fold(
+          _ => Future.successful(Redirect(routes.MissingInformationController.onPageLoad(operatorId))),
+          submissionRequest => (for {
+            submission <- submit(submissionRequest.asAssumedReportingSubmissionRequest, platformOperator)
+            summary <- AssumedReportSummary(request.userAnswers).map(Future.successful).getOrElse(Future.failed(Exception("unable to build an assumed report summary")))
```

Comment:
Can we add a test when the Future.failed exception is thrown for "unable to build an assumed report summary"